crates/storage/provider/src/providers/static_file/writer.rs

```diff
@@ -547,6 +547,39 @@ impl StaticFileProviderRW {
         Ok(result)
     }
 
+    /// Appends multiple receipts to the static file.
+    ///
+    /// Returns the current [`TxNumber`] as seen in the static file.
+    pub fn append_receipts<I>(&mut self, receipts: I) -> ProviderResult<TxNumber>
+    where
+        I: IntoIterator<Item = (TxNumber, Receipt)>,
+    {
+        let mut receipts_iter = receipts.into_iter().peekable();
+        if receipts_iter.peek().is_none() {
+            return Err(ProviderError::EmptyReceipts());
+        }
+
+        let start = Instant::now();
+        self.ensure_no_queued_prune()?;
+        // receipts contains at least one receipt so this would be overwritten
+        let mut last_tx_number = TxNumber::default();
+
+        for (tx_num, receipt) in receipts_iter {
+            self.append_with_tx_number(StaticFileSegment::Receipts, tx_num, receipt)?;
+            last_tx_number = tx_num;
+        }
+
+        if let Some(metrics) = &self.metrics {
+            metrics.record_segment_operation(
+                StaticFileSegment::Receipts,
+                StaticFileProviderOperation::Append,
+                Some(start.elapsed()),
+            );
```

Comment:
this will report the time it took to insert multiple receipts, but above we use this metric to report the time for one receipt. Let's introduce `record_segment_operations` method that will accept the number of operations and the total time spent, and it will report the average time per operation.